plugins/wfp_auth/views.py

```diff
@@ -111,7 +112,11 @@ def complete_login(self, request, app, token: str, response) -> SocialAccount:
             email = extra_data["sub"].lower().strip()
         # the sub is the email, wfp verify it so let's trust this
         uid = extra_data["sub"].lower().strip()
-        account = Account.objects.get(name=self.settings["IASO_ACCOUNT_NAME"])
+        app_id = AppIdSerializer(data=self.request.query_params).get_app_id(raise_exception=False)
+        if app_id:
+            account = Project.objects.get(app_id=app_id).account
```

Comment:
In fact, at this point, we should do 
project = get_object_or_404(Project, app_id=app_id)
account = project.account 

so that the API returns a 404 and not a 500 if the provided app_id is wrong

Reply:
(sorry, I know, I gave the wrong code before)

Reply:
The import to do: from django.shortcuts import get_object_or_404

Reply:
I have pushed the requested change.